### PR TITLE
[Refactor] Add reduction_override in MSELoss

### PR DESCRIPTION
**Pull request by BIGWangYuDong**
Add reduction_override in MSELoss.

When using MSELoss replace L1Loss in Faster R-CNN or Mask R-CNN, may get a bug, this PR adds reduction_override Args in MSELoss.

### CODE REVIEW

**Comment by ZwwWayne** [comment]
Need unit tests.

**Review by jshilong — APPROVED** [review]

**Comment by jshilong** [comment]
add unit tests especially for the `override`

File: tests/test_models/test_loss.py [hunk 1]
```diff
@@ -14,3 +17,18 @@ def test_iou_type_loss_zeros_weight(loss_class):
...
+    IoULoss, BoundedIoULoss, GIoULoss, DIoULoss, CIoULoss, MSELoss, L1Loss,
+    SmoothL1Loss, BalancedL1Loss, FocalLoss, QualityFocalLoss,
+    GaussianFocalLoss, DistributionFocalLoss, VarifocalLoss
+])
+def test_loss_with_reduction_override(loss_class):
+    pred = torch.rand((10, 4))
+    target = torch.rand((10, 4))
+
+    with pytest.raises(AssertionError):
+        # assert whether reduction_override in loss function
```
Comment (ZwwWayne) [inline tests/test_models/test_loss.py:32]: # only reduction_override from [None, 'none', 'mean', 'sum'] is not allowed

File: tests/test_models/test_loss.py [hunk 2]
```diff
@@ -14,3 +17,18 @@ def test_iou_type_loss_zeros_weight(loss_class):
...
+
+@pytest.mark.parametrize('loss_class', [
+    IoULoss, BoundedIoULoss, GIoULoss, DIoULoss, CIoULoss, MSELoss, L1Loss,
+    SmoothL1Loss, BalancedL1Loss, FocalLoss, QualityFocalLoss,
+    GaussianFocalLoss, DistributionFocalLoss, VarifocalLoss
+])
+def test_loss_with_reduction_override(loss_class):
+    pred = torch.rand((10, 4))
+    target = torch.rand((10, 4))
+
```
Comment (ZwwWayne) [inline tests/test_models/test_loss.py:30]: We also need to test the normal case

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**